devnet-topo.sh

```diff
@@ -0,0 +1,192 @@
+total_clients=${total_validators}
+
+# Read the network ID from user or use a default value of 1
+read -p "Enter the network ID (mainnet = 0, testnet = 1, canary = 2) (default: 1): " network_id
+network_id=${network_id:-1}
+
+# Ask the user if they want to run 'cargo install --locked --path .' or use a pre-installed binary
+read -p "Do you want to run 'cargo install --locked --path .' to build the binary? (y/n, default: y): " build_binary
+build_binary=${build_binary:-y}
+
+# Ask the user whether to clear the existing ledger history
+read -p "Do you want to clear the existing ledger history? (y/n, default: n): " clear_ledger
+clear_ledger=${clear_ledger:-n}
+
+# Log verbosity is set to 1 (DEBUG) by default.
+verbosity=1
+
+if [[ $build_binary == "y" ]]; then
+  # Ask the user if they want to enable validator telemetry
+  read -p "Do you want to enable validator telemetry? (y/n, default: y): " enable_telemetry
+  enable_telemetry=${enable_telemetry:-y}
+
+  # Build command
+  build_cmd="cargo install --locked --path ."
+
+  # Add the telemetry feature if requested
+  if [[ $enable_telemetry == "y" ]]; then
+    build_cmd+=" --features telemetry"
+  fi
+
+  # Build command
+  echo "Running build command: \"$build_cmd\""
+  eval "$build_cmd" || exit 1
+fi
+
+# Clear the ledger logs for each validator if the user chooses to clear ledger
+if [[ $clear_ledger == "y" ]]; then
+  # Create an array to store background processes
+  clean_processes=()
+
+  for ((index = 0; index < $((total_validators + total_clients)); index++)); do
+    # Run 'snarkos clean' for each node in the background
+    snarkos clean --network $network_id --dev $index &
+
+    # Store the process ID of the background task
+    clean_processes+=($!)
+  done
+
+  # Wait for all 'snarkos clean' processes to finish
+  for process_id in "${clean_processes[@]}"; do
+    wait "$process_id"
+  done
+fi
+
+# Create a timestamp-based directory for log files
+log_dir=".logs-$(date +"%Y%m%d%H%M%S")"
+mkdir -p "$log_dir"
+
+# Create a new tmux session named "devnet"
+tmux new-session -d -s "devnet" -n "validator-0"
+if [[ $? -ne 0 ]]; then
+   echo "Failed to create new TMUX session."
+   exit 1
+fi
+
+# Get the tmux's base-index for windows
+# we have to create all windows with index offset by this much
+index_offset="$(tmux show-option -gv base-index)"
+if [ -z "$index_offset" ]; then
+  index_offset=0
+fi
+
+# Generate validator indices from 0 to (total_validators - 1).
+validator_indices=($(seq 0 $((total_validators - 1))))
+
+# Loop through the list of validator indices and create a new window for each
+for validator_index in "${validator_indices[@]}"; do
+  # Generate a unique and incrementing log file name based on the validator index
+  name="validator-$validator_index"
+  log_file="$log_dir/$name.log"
+  window_index=$((validator_index + index_offset))
+  metrics_flag=""
+
+  if [ "$validator_index" -eq 0 ]; then
+    # Enable metrics only for the first validator.
```

Comment:
We can enable metrics for all validators, like we do in the devnet.sh script now.